// Validating benchmarks:
// ***** BenchmarkRunner: Start   *****
// ***** Found 2 benchmark(s) in total *****
// ***** Building 1 exe(s) in Parallel: Start   *****
// start dotnet restore  /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true in C:\Users\Aleksandr\source\repos\Task1\TestProject\bin\Release\net5.0\8a0434b2-9827-4df0-b070-63cef09b4527
// command took 1,81s and exited with 0
// start dotnet build -c Release  --no-restore /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true in C:\Users\Aleksandr\source\repos\Task1\TestProject\bin\Release\net5.0\8a0434b2-9827-4df0-b070-63cef09b4527
// command took 3,58s and exited with 0
// ***** Done, took 00:00:05 (5.49 sec)   *****
// Found 2 benchmarks:
//   Benchmark.MultiThread: DefaultJob
//   Benchmark.Sequential: DefaultJob

// **************************
// Benchmark: Benchmark.MultiThread: DefaultJob
// *** Execute ***
// Launch: 1 / 1
// Execute: dotnet "8a0434b2-9827-4df0-b070-63cef09b4527.dll" --benchmarkName "TestProject.Benchmark.MultiThread" --job "Default" --benchmarkId 0 in C:\Users\Aleksandr\source\repos\Task1\TestProject\bin\Release\net5.0\8a0434b2-9827-4df0-b070-63cef09b4527\bin\Release\net5.0
// BeforeAnythingElse

// Benchmark Process Environment Information:
// Runtime=.NET 5.0.2 (5.0.220.61120), X64 RyuJIT
// GC=Concurrent Workstation
// Job: DefaultJob

OverheadJitting  1: 1 op, 294900.00 ns, 294.9000 us/op

System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at Task1.Prime.HasNonPrimeNumberMultiThread(Int32[] array, Byte threadCount) in C:\Users\Aleksandr\source\repos\Task1\Task1\Prime.cs:line 55
   at TestProject.Benchmark.MultiThread() in C:\Users\Aleksandr\source\repos\Task1\TestProject\Benchmark.cs:line 26
   at BenchmarkDotNet.Autogenerated.Runnable_0.WorkloadActionNoUnroll(Int64 invokeCount) in C:\Users\Aleksandr\source\repos\Task1\TestProject\bin\Release\net5.0\8a0434b2-9827-4df0-b070-63cef09b4527\8a0434b2-9827-4df0-b070-63cef09b4527.notcs:line 842
   at BenchmarkDotNet.Engines.Engine.RunIteration(IterationData data)
   at BenchmarkDotNet.Engines.EngineFactory.Jit(Engine engine, Int32 jitIndex, Int32 invokeCount, Int32 unrollFactor)
   at BenchmarkDotNet.Engines.EngineFactory.CreateReadyToRun(EngineParameters engineParameters)
   at BenchmarkDotNet.Autogenerated.Runnable_0.Run(IHost host, String benchmarkName) in C:\Users\Aleksandr\source\repos\Task1\TestProject\bin\Release\net5.0\8a0434b2-9827-4df0-b070-63cef09b4527\8a0434b2-9827-4df0-b070-63cef09b4527.notcs:line 168
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at BenchmarkDotNet.Autogenerated.UniqueProgramName.AfterAssemblyLoadingAttached(String[] args) in C:\Users\Aleksandr\source\repos\Task1\TestProject\bin\Release\net5.0\8a0434b2-9827-4df0-b070-63cef09b4527\8a0434b2-9827-4df0-b070-63cef09b4527.notcs:line 52
// AfterAll
// Benchmark Process 31344 has exited with code -1.
// * Artifacts cleanup *
